cpp/arcticdb/version/version_map.hpp

```diff
@@ -512,7 +506,7 @@ class VersionMapImpl {
         auto journal_key = to_atom(std::move(journal_single_key(store, key, entry->head_)));
         write_to_entry(entry, key, journal_key);
         auto previous_index = entry->get_second_undeleted_index();
-        write_symbol_ref(store, key, previous_index, journal_key);
+        return previous_index;
```

Comment:
This seems logically wrong. `do_write` is called from a few places, some of which then call `write_symbol_ref`, and some of which don't.

Reply:
OK, you've convinced me 😄